lib/postgrex/extensions/binary.ex

```diff
@@ -162,10 +162,15 @@ defmodule Postgrex.Extensions.Binary do
       bin = for digit <- digits, into: "", do: <<digit :: uint16>>
       ndigits = div(byte_size(bin), 2)
 
-      [<<ndigits :: int16, weight :: int16, sign :: uint16, scale :: int16>>, bin]
```

Comment:
The new style will copy the binary, while using a list will just reference to it. In fact, the ideal return memory and cpu wise would be `[<<...>>|bin]`.